cpp/examples/arrow/dataset_parquet_scan_example.cc

```diff
@@ -62,8 +73,11 @@ struct Configuration {
 
   // Indicates the filter by which rows will be filtered. This optimization can
   // make use of partition information and/or file metadata if possible.
-  cp::Expression filter =
-      cp::greater(cp::field_ref("total_amount"), cp::literal(1000.0f));
+  // Equivalent filter with field_name instead of field_index
+  // cp::Expression filter = cp::greater(cp::field_ref("total_amount"),
+  // cp::literal(1000.0f));
+  // 0: pickup_at, 1: dropoff_at, 2: total_amount
+  cp::Expression filter = cp::greater(cp::field_ref(2), cp::literal(1000.0f));
```

Comment:
I think this comment makes the example a little more complex than needed.  Maybe we could do an AND expression `field_ref(2) > 1000 & field_ref("total_amount") < 2000` to demonstrate both types of field refs?  Or we could just leave it how it was and not worry about an example field ref by id.

Reply:
This is indeed a good one. Adding that.